Framework/API/src/ExperimentInfo.cpp

```diff
@@ -297,14 +304,30 @@ void ExperimentInfo::setInstrument(const Instrument_const_sptr &instr) {
       sptr_instrument, m_parmap);
   m_detectorInfo = makeDetectorInfo(*parInstrument, *instr);
   m_parmap->setDetectorInfo(m_detectorInfo);
-  m_detectorInfoWrapper = Kernel::make_unique<DetectorInfo>(
-      *m_detectorInfo, getInstrument(), m_parmap.get());
+  if (instr->hasDetectorInfo()) {
 
-  std::tie(m_componentInfo, m_componentIds) =
+    // Reuse the ID -> index map for the detector ids
+    m_detectorInfoWrapper = Kernel::make_unique<DetectorInfo>(
+        *m_detectorInfo, getInstrument(), m_parmap.get(),
+        instr->detIdToIndexMap());
+  } else {
+    m_detectorInfoWrapper = Kernel::make_unique<DetectorInfo>(
+        *m_detectorInfo, getInstrument(), m_parmap.get(),
+        makeDetIdToIndexMap(instr->getDetectorIDs()));
```

Comment:
Yes. That's correct it's now redundant. It needs to be removed. I was going to do this as a latter step so that we could get the fix out and available for Vickie asap.